Facing compilation error due to the below attached warning on both v1.0.4 and v2.0.0

[hramesh@C8K69D3:synced] >make
Makefile:28: ESMC_STACK is not defined
Makefile:30: Setting ESMC_STACK to renesas (default)...
Makefile:37: PLATFORM is not defined
Makefile:39: Setting PLATFORM to amd64 (default)...
Makefile:43: CROSS_COMPILE is not defined
Makefile:54: DEVICE is not defined
Makefile:56: Setting DEVICE to generic (default)...
Makefile:59: SYNCED_DEBUG_MODE is not defined (SYNCED_DEBUG_MODE must be either 0 or 1)
Makefile:61: Setting SYNCED_DEBUG_MODE to 0 (default)...
#################################################
#
# C L E A N I N G   B U I L D   A R T I F A C T S
#
#################################################
rm -f -r build/bin/synced
rm -f -r build/bin/synced_cli
rm -f -rf build/obj
rm -f -rf build/obj/management/cli
rm -f -rf pkg
###############################
#
# B U I L D I N G   S Y N C E D
#
###############################
ESMC_STACK: renesas
PLATFORM: amd64
CROSS_COMPILE:
DEVICE: generic
SYNCED_DEBUG_MODE: 0
mkdir -p build/bin
mkdir -p build/obj
mkdir -p build/obj/management/cli
mkdir -p build/obj/common/
gcc \
        -I. -Icommon -Icontrol -Idevice/generic -Idevice/device_adaptor -Idevice/generic -Idevice/rsmu -Iesmc/esmc_adaptor -Iesmc/renesas -Imanagement -Imonitor \
        -DDEVICE=generic -DSYNCED_DEBUG_MODE=0 -Wall -Werror -Wpedantic -Wextra \
        -o build/obj/common/common.o \
        -c common/common.c
mkdir -p build/obj/common/
gcc \
        -I. -Icommon -Icontrol -Idevice/generic -Idevice/device_adaptor -Idevice/generic -Idevice/rsmu -Iesmc/esmc_adaptor -Iesmc/renesas -Imanagement -Imonitor \
        -DDEVICE=generic -DSYNCED_DEBUG_MODE=0 -Wall -Werror -Wpedantic -Wextra \
        -o build/obj/common/config.o \
        -c common/config.c
common/config.c: In function ‘config_read’:
common/config.c:654:67: error: argument to ‘sizeof’ in ‘strncpy’ call is the same expression as the source; did you mean to use the size of the destination? [-Werror=sizeof-pointer-memaccess]
         strncpy(g_config_port_name[port_counter - 1], port, sizeof(port));
                                                                   ^
cc1: all warnings being treated as errors
make: *** [Makefile:244: build/obj/common/config.o] Error 1

**PS:** 
-->gcc version used 
[hramesh@C8K69D3:synced] >gcc --version
gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-10)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

--- a/common/config.c
+++ b/common/config.c
@@ -651,7 +651,7 @@ int config_read(const char *name,
           goto parse_error;
         }
 
-        strncpy(g_config_port_name[port_counter - 1], port, sizeof(port));
+        strncpy(g_config_port_name[port_counter - 1], port, strlen(port) + 1);
       }
 
       continue;

